EIPS/eip-6110.md

```diff
@@ -0,0 +1,192 @@
+
+```python
+def compute_trie_root_from_indexed_data(data):
+    trie = Trie.from([(i, obj) for i, obj in enumerate(data)])
+    return trie.root
+
+block.header.deposits_root = compute_trie_root_from_indexed_data(block.body.deposits)
+```
+
+### Block validity
+
+Beginning with the `FORK_BLOCK`, client software **MUST** extend block validity rule set with the following conditions:
+
+1. Value of **`deposits_root`** block header field equals to the trie root committing to the list of deposit operations contained in the block. To illustrate:
+
+```python
+def compute_trie_root_from_indexed_data(data):
+    trie = Trie.from([(i, obj) for i, obj in enumerate(data)])
+    return trie.root
+
+assert block.header.deposits_root == compute_trie_root_from_indexed_data(block.body.deposits)
+```
+
+2. The list of deposit operations contained in the block **MUST** be equivalent to the list of log events emitted by each deposit transaction of the given block, respecting the order of transaction inclusion. To illustrate:
+
+```python
+def parse_deposit_data(deposit_event_data): bytes[]
+  """
+  Parses Deposit operation from DepositContract.DepositEvent data
+  """
+  pass
+
+def little_endian_to_uint64(data: bytes): uint64
+    return uint64(int.from_bytes(data, 'little'))
+
+class Deposit(object):
+    pubkey: Bytes48
+    withdrawal_credentials: Bytes32
+    amount: uint64
+    signature: Bytes96
+    index: uint64
+
+def event_data_to_deposit(deposit_event_data): Deposit
+  deposit_data = parse_deposit_data(deposit_event_data)
+  return Deposit(
+    pubkey=Bytes48(deposit_data[0]),
+    withdrawal_credentials=Bytes32(deposit_data[1]),
+    amount=little_endian_to_uint64(deposit_data[2]),
+    signature=Bytes96(deposit_data[3]),
+    index=little_endian_to_uint64(deposit_data[4])
+  )
+
+# Obtain receipts from block execution result
+receipts = block.execution_result.receipts
+
+# Retrieve all deposits made in the block
+expected_deposits = []
+for receipt in receipts:
+    for log in receipt.logs:
+        if log.address == DEPOSIT_CONTRACT_ADDRESS:
+            deposit = event_data_to_deposit(log.data)
+            expected_deposits.append(deposit)
+
+# Compare retrieved deposits to the list in the block body
+assert block.body.deposits == expected_deposits
+```
+
+A block that does not satisfy the above conditions **MUST** be deemed invalid.
+
+## Rationale
+
+### `index` field
+
+The `index` field may appear unnecessary but it is important information for deposit processing flow on the Consensus Layer.
+
+### Not limiting the size of deposit operations list
+
+The list is unbounded because of negligible data complexity and absence of potential DoS vectors. See [Security Considerations](#Security-Considerations) for more details.
```

Comment:
Yeah, I guess this is *not* an EL security consideration because it's just data. on CL side, the signature processing (if done in sync) will be a security consideration

Reply:
I agree that *more* considerations should be taken on CL side, but on EL it is still a potential DoS vector on the networking layer which analysis is relevant to the security section, IMO.